test/Test/Time/Property.hs

```diff
@@ -88,17 +87,15 @@ rationalNum = do
                              else natural 1
     return $ numeratorVal % denomVal
 
-numRatPair :: (Monad m) => PropertyT m (Int, RatioNat)
-numRatPair = liftA2 (,) (forAll number0'9) (forAll rationalNum)
+anyTime :: (Monad m) => PropertyT m AnyTime
+anyTime = do
+    n      <- forAll number0'9
+    ratNum <- forAll rationalNum
+    pure $ unitChooser n ratNum
 
 -- | Property test.
 prop_readShowUnit :: Property
-prop_readShowUnit = property $ do
-    (n, ratN) <- numRatPair
-    verifyAnyTime $ unitChooser n ratN
+prop_readShowUnit = property $ anyTime >>= verifyAnyTime
 
 prop_toUnit :: Property
-prop_toUnit = property $ do
-    (n1, ratN1) <- numRatPair
-    (n2, ratN2) <- numRatPair
-    verifyToUnit (unitChooser n1 ratN1) (unitChooser n2 ratN2)
+prop_toUnit = property $ anyTime >>= \t1 -> anyTime >>= verifyToUnit t1
```

Comment:
This is not readable to me. Mix of `>>=` with variable names and partial application doesn't really improves things... This looks better to me:

```haskell
t1 <- anyTime
t2 <- anyTime
verityToUnit t1 t2
```

To write this shorter you need combinator like this one

```haskell
Monad m => (a -> b -> m c) -> m a -> m b -> m c
```
It's very useful but I'm not aware of such combinators :(
So I propose to write this in `do`-notation.